scipy/linalg/basic.py

```diff
@@ -777,26 +777,48 @@ def lstsq(a, b, cond=None, overwrite_a=False, overwrite_b=False,
         Whether to check that the input matrices contain only finite numbers.
         Disabling may give a performance gain, but may result in problems
         (crashes, non-termination) if the inputs do contain infinities or NaNs.
+    lapack_driver: str, optional
+        Which LAPACK driver is used to solve the least-squares problem.
+        Options are ``'gelsd'``, ``'gelsy'``, ``'gelss'``. Default
+        (``'gelsd'``) is a good choice.  However, ``'gelsy'`` can be slightly
+        faster on many problems.  ``'gelss'`` was used historically.  It is
+        generally slow but uses less memory.
+
+        .. versionadded:: 0.17.0
 
     Returns
     -------
+    If lapack_driver is ``'gelsd'`` or ``'gelss'``:
+
     x : (N,) or (N, K) ndarray
         Least-squares solution.  Return shape matches shape of `b`.
     residues : () or (1,) or (K,) ndarray
         Sums of residues, squared 2-norm for each column in ``b - a x``.
-        If rank of matrix a is < N or > M this is an empty array.
+        If rank of matrix a is ``< N`` or ``> M``, this is an empty array.
         If b was 1-D, this is an (1,) shape array, otherwise the shape is (K,).
     rank : int
         Effective rank of matrix `a`.
     s : (min(M,N),) ndarray
         Singular values of `a`. The condition number of a is
-        ``abs(s[0]/s[-1])``.
+        ``abs(s[0] / s[-1])``.
+
+    If lapack_driver is 'gelsy':
+
+    x : (N,) or (N, K) ndarray
+        Least-squares solution.  Return shape matches shape of `b`.
+    j : (N,) ndarray of integer
```

Comment:
Are the pivots of any use if the rest of the QR factorization is not returned?
If not, probably best to not return them (users can always call the LAPACK function themselves).